models/amazon_selling_partner__order_items.sql

```diff
@@ -0,0 +1,116 @@
+{{ config(enabled=var('amazon_selling_partner__using_orders_module', true)) }}
+
+with orders as (
+
+    select *
+    from {{ ref('stg_amazon_selling_partner__orders') }}
+),
+
+order_item as (
+
+    select *
+    from {{ ref('stg_amazon_selling_partner__order_item') }}
+),
+
+{% if var('amazon_selling_partner__using_catalog_module', true) %}
+{# Open Q: Should we bring item in here?  #}
+item as (
+
+    select *
+    from {{ ref('int_amazon_selling_partner__item') }}
+),
+{% endif %}
+
+order_item_promotion_id as (
+    
+    select *
+    from {{ ref('stg_amazon_selling_partner__order_item_promotion_id') }}
+),
+
+aggregate_promotions as (
+
+    select 
+        source_relation,
+        amazon_order_id,
+        order_item_id,
+        count(distinct promotion_id) as count_promotions_used
+
+    from order_item_promotion_id 
+    group by 1,2,3
+),
+
+joined as (
+
+    select 
+        order_item.*,
+        {# Open Q: What other order/header level fields would be useful to have here? #}
+        orders.purchase_date as order_purchase_date,
+        orders.order_total_amount, 
+        orders.order_status,
+        orders.order_total_currency_code,
+        orders.marketplace_id,
+
+        aggregate_promotions.count_promotions_used,
+
+        {# Open Q: Which item columns (if any) should we bring in here? #}
```

Comment:
I would lean more towards a few key columns as opposed to an exhaustive list. If users want more then they can request additional columns to be added. My recommendation is to include the following as they seem like relevant fields when analyzing orders:
```sql
        item.item_name,
        item.display_name,
        item.brand,
        item.color,
        item.size,
        item.style,
        item.product_type,
        item.package_quantity,
        item.item_classification,
```

Reply:
sounds good, limited